File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260917-060430-47161/workdir/src/main/java/surveycrawler/NormalRun.java
package surveycrawler;

public class NormalRun extends CrawlerRun {

    // guarded run

    @Override
    public void startGuardedRun(SurveyCrawler aCrawler) {
        aCrawler.runGuarded();
    }

    @Override
    public void finishGuardedRun(SurveyCrawler aCrawler) {
        aCrawler.signalGuardedRunNotStarted();
    }

    @Override
    public void assertFinished(SurveyCrawler aCrawler) {
    }

    // moving

    @Override
    public void siltReached(SurveyCrawler aCrawler) {
    }

    // undoing

    @Override
    public void recordUndo(Runnable anUndoMovement) {
    }

    @Override
    public void undo() {
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-204513-77784/analysis/own-tests-project/src/surveycrawler/normal_run.py
from surveycrawler.crawler_run import CrawlerRun


class NormalRun(CrawlerRun):

    # command processing

    def start_guarded_run(self, a_crawler):
        a_crawler.start_guarded_run()

    def finish_guarded_run(self, a_crawler):
        a_crawler.signal_guarded_run_not_started()

    def assert_is_finished(self, a_crawler):
        pass

    # moving

    def slide_crawler_towards(self, a_crawler, a_direction, a_silt):
        a_crawler.move_cells_towards(a_silt.number_of_cells_to_slide(), a_direction)

    # undoing

    def register_movement(self, a_movement):
        pass

    def undo_movements(self, a_crawler):
        pass
